app/controllers/saml_idp_controller.rb

```diff
@@ -94,18 +109,17 @@ def process_invalid_authn_context
     render nothing: true, status: :bad_request
   end
 
-  def authn_context_node
-    saml_request.document.xpath(
-      '//samlp:AuthnRequest/samlp:RequestedAuthnContext/saml:AuthnContextClassRef',
-      samlp: Saml::XML::Namespaces::PROTOCOL,
-      saml: Saml::XML::Namespaces::ASSERTION)
+  def saml_request_document
+    @_saml_xml_document ||= Saml::XML::Document.parse(saml_request.raw_xml)
```

Comment:
This is a new method that has been added, but I don't see it being called anywhere in this PR. Is it needed?